### PR TITLE
klipper-firmware: use gcc-arm-embedded-11

### PR DESCRIPTION
**Pull request by vtuan10**
## Description of changes
Workaround for https://github.com/NixOS/nixpkgs/issues/263370 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/klipper/klipper-firmware.nix [hunk 1]
```diff
@@ -7,7 +7,7 @@
 , libusb1
 , wxGTK32
 , python3
-, gcc-arm-embedded
+, gcc-arm-embedded-11
```
Comment (arachnist) [inline pkgs/servers/klipper/klipper-firmware.nix:10]: Wouldn't it be better to override this in `all-packages.nix` like this?

```diff
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index eacb266e2ff7..f039b169a82c 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5783,7 +5783,7 @@ with pkgs;
 
   klipper = callPackage ../servers/klipper { };
 
-  klipper-firmware = callPackage ../servers/klipper/klipper-firmware.nix { };
+  klipper-firmware = callPackage ../servers/klipper/klipper-firmware.nix { gcc-arm-embedded = gcc-arm-embedded-11; };
 
   klipper-flash = callPackage ../servers/klipper/klipper-flash.nix { };
```

Comment (vtuan10) [reply pkgs/servers/klipper/klipper-firmware.nix]: Yeah makes more sense. Updated it

Comment (arachnist) [reply pkgs/servers/klipper/klipper-firmware.nix]: @vtuan10 I don't have too much experience with packaging stuff in nixpkgs. Both your method and mine fix this particular issue, but I don't see myself as an authority to say which one is better in any way, or why.

Comment (vtuan10) [reply pkgs/servers/klipper/klipper-firmware.nix]: Your way is better IMO as this is a current workaround until klipper fixes this issue, no need to change the `klipper-firmware` derivation 👍

**Review by arachnist — APPROVED** [review]

**Pull request merged**